File: estruturas_de_dados_e_algoritmos/1 - topicos iniciais/python/program.py
class Product:
    def __init__(self, name, price, quantity):
        self.name = name
        self.price = price
        self.quantity = quantity
    
    def __str__(self):
        return f"{self.name}, ${self.price:.2f}, {self.quantity}"
    
    def total(self):
        return self.price * self.quantity
    
    def update_price(self, percentage):
        self.price = self.price + (1 + percentage / 100)
    
def total(product):
    return product.price * product.quantity


def update_price(product, percentage):
    product.price = product.price * (1 + percentage / 100 )

p1 = Product("P1", 1000.0, 5)

print(total(p1))
update_price(p1, 10)
print(p1.price)

p1.total()
p1.update_price(10)

File: estruturas_de_dados_e_algoritmos/1 - topicos iniciais/csharp/Program.cs
namespace Course 
{
    internal class Program 
    {

        static double total(Product product)
        {
            return Product.Price * Product.Quantity;
        }

        static void updatePrice(Product product, Double percentage)
        {
            Product.Price = Product.Price * (1.0 + percentage / 100.0);
        }

        static void Main(String[] args)
        {
            var p1 = new Product("P1", 1000.0, 5);
            Console.WriteLine(p1);
            Console.WriteLine(total(p1));
            Console.WriteLine(updatePrice(p1, 10));

            p1.total();
            p1.updatePrice();
        }
    }

    class Product
    {
        public string Name { get; set; }
        public double Price { get; set; }
        public int Quantity { get; set; }

        public Product(string name, double price, int Quantity ) {
            this.Name = name;
            this.Price = price;
            this.Quantity = Quantity;
        }

        public double total()
        {
            return this.Price * this. Quantity;
        }

        public void updatePrice(Double percentage)
        {
            this.Price = this.Price * (1.0 + percentage / 100.0);
        }

        public override string ToString()
        {
            return Name + ", " + Price + ", "   + Quantity;
        }
    }
}